embeddings/embedding/static/flair.py

```diff
@@ -0,0 +1,159 @@
+
+        elif embeddings.lower() in ["pubmed", "en-pubmed"]:
+            cached_path(
+                f"{hu_path}/pubmed_pmc_wiki_sg_1M.gensim.vectors.npy",
+                cache_dir=cache_dir,
+            )
+            embeddings_path = cached_path(f"{hu_path}/pubmed_pmc_wiki_sg_1M.gensim", cache_dir=cache_dir)
+
+        elif embeddings.lower() in ["crawl", "en-crawl"]:
+            cached_path(f"{hu_path}/en-fasttext-crawl-300d-1M.vectors.npy", cache_dir=cache_dir)
+            embeddings_path = cached_path(f"{hu_path}/en-fasttext-crawl-300d-1M", cache_dir=cache_dir)
+
+        elif embeddings.lower() in ["news", "en-news", "en"]:
+            cached_path(f"{hu_path}/en-fasttext-news-300d-1M.vectors.npy", cache_dir=cache_dir)
+            embeddings_path = cached_path(f"{hu_path}/en-fasttext-news-300d-1M", cache_dir=cache_dir)
+
+        elif embeddings.lower() in ["twitter", "en-twitter"]:
+            cached_path(f"{hu_path}/twitter.gensim.vectors.npy", cache_dir=cache_dir)
+            embeddings_path = cached_path(f"{hu_path}/twitter.gensim", cache_dir=cache_dir)
+
+        elif len(embeddings.lower()) == 2:
+            cached_path(
+                f"{hu_path}/{embeddings}-wiki-fasttext-300d-1M.vectors.npy",
+                cache_dir=cache_dir,
+            )
+            embeddings_path = cached_path(f"{hu_path}/{embeddings}-wiki-fasttext-300d-1M", cache_dir=cache_dir)
+
+        elif len(embeddings.lower()) == 7 and embeddings.endswith("-wiki"):
+            cached_path(
+                f"{hu_path}/{embeddings[:2]}-wiki-fasttext-300d-1M.vectors.npy",
+                cache_dir=cache_dir,
+            )
+            embeddings_path = cached_path(f"{hu_path}/{embeddings[:2]}-wiki-fasttext-300d-1M", cache_dir=cache_dir)
+
+        elif len(embeddings.lower()) == 8 and embeddings.endswith("-crawl"):
+            cached_path(
+                f"{hu_path}/{embeddings[:2]}-crawl-fasttext-300d-1M.vectors.npy",
+                cache_dir=cache_dir,
+            )
+            embeddings_path = cached_path(
+                f"{hu_path}/{embeddings[:2]}-crawl-fasttext-300d-1M",
+                cache_dir=cache_dir,
+            )
+
+        elif not Path(embeddings).exists():
+            raise ValueError(f'The given embeddings "{embeddings}" is not available or is not a valid path.')
+        else:
+            embeddings_path = Path(embeddings)
+
+        self.name = str(embeddings_path)
+        self.static_embeddings = not fine_tune
+        self.fine_tune = fine_tune
+        self.force_cpu = force_cpu
+        self.field = field
+        self.stable = stable
+
+        if embeddings_path.suffix == ".bin":
+            precomputed_word_embeddings = gensim.models.KeyedVectors.load_word2vec_format(
+                str(embeddings_path), binary=True
+            )
+        else:
+            try:
+                precomputed_word_embeddings = gensim.models.KeyedVectors.load(str(embeddings_path))
+            except UnpicklingError:
```

Comment:
It's often good practice to log (as a warning for example) events when they get to the `except` clause. I don't know if it applies here, but something to consider 👌